sherpa/optmethods/optfcts.py

```diff
@@ -122,78 +124,52 @@ def _get_saofit_msg(maxfev, ierr):
 
 
 def _move_within_limits(x, xmin, xmax):
-    below = numpy.flatnonzero(x < xmin)
+    below = np.flatnonzero(x < xmin)
     if below.size > 0:
         x[below] = xmin[below]
 
-    above = numpy.flatnonzero(x > xmax)
+    above = np.flatnonzero(x > xmax)
     if above.size > 0:
         x[above] = xmax[above]
 
 
-def _my_is_nan(x):
-    fubar = list(filter(lambda xx: xx != xx or xx is numpy.nan or numpy.isnan(xx) and numpy.isfinite(xx), x))
-    return len(fubar) > 0
-
-
-def _narrow_limits(myrange, xxx, debug):
-
-    def double_check_limits(myx, myxmin, myxmax):
-        for my_l, my_x, my_h in zip(myxmin, myx, myxmax):
-            if my_x < my_l:
-                print('x = ', my_x, ' is < lower limit = ', my_l)
-            if my_x > my_h:
-                print('x = ', my_x, ' is > upper limit = ', my_h)
-
-    def raise_min_limit(xrange, xmin, x, debug=False):
-        myxmin = numpy.asarray(list(map(lambda xx: xx - xrange * numpy.abs(xx), x)), numpy.float64)
-        if debug:
-            print()
-            print(f'raise_min_limit: myxmin={myxmin}')
-            print(f'raise_min_limit: x={x}')
-        below = numpy.flatnonzero(myxmin < xmin)
-        if below.size > 0:
-            myxmin[below] = xmin[below]
-        if debug:
-            print(f'raise_min_limit: myxmin={myxmin}')
-            print(f'raise_min_limit: x={x}')
-            print()
-        return myxmin
-
-    def lower_max_limit(xrange, x, xmax, debug=False):
-        myxmax = numpy.asarray(list(map(lambda xx: xx + xrange * numpy.abs(xx), x)), numpy.float64)
-        if debug:
-            print()
-            print(f'lower_max_limit: x={x}')
-            print(f'lower_max_limit: myxmax={myxmax}')
-        above = numpy.flatnonzero(myxmax > xmax)
-        if above.size > 0:
-            myxmax[above] = xmax[above]
-        if debug:
-            print(f'lower_max_limit: x={x}')
-            print(f'lower_max_limit: myxmax={myxmax}')
-            print()
-        return myxmax
-
-    x = xxx[0]
-    xmin = xxx[1]
-    xmax = xxx[2]
-
-    if debug:
-        print(f'narrow_limits: xmin={xmin}')
-        print(f'narrow_limits: x={x}')
-        print(f'narrow_limits: xmax={xmax}')
-    myxmin = raise_min_limit(myrange, xmin, x, debug=False)
-    myxmax = lower_max_limit(myrange, x, xmax, debug=False)
-
-    if debug:
-        print(f'range = {myrange}')
-        print(f'narrow_limits: myxmin={myxmin}')
-        print(f'narrow_limits: x={x}')
-        print(f'narrow_limits: myxmax={myxmax}\n')
-
-    double_check_limits(x, myxmin, myxmax)
+def _raise_min_limit(factor: float,
+                     xmin: np.ndarray,
+                     x: np.ndarray
+                     ) -> np.ndarray:
+    """Calculate the new minimum limits."""
 
+    myxmin = x - factor * np.abs(x)
+    below = np.flatnonzero(myxmin < xmin)
+    if below.size > 0:
+        myxmin[below] = xmin[below]
```

Comment:
```suggestion
    below = myxmin < xmin
    myxmin[below] = xmin[below]
```

Reply:
Yes, even better. From 3 lines to 2 lines to one line. More readable!

Reply:
So I've made these changes, as well as in `_move_within_limits` which then becomes a one-liner so can get put directly into the one routine that called it.